src/tracing_library.cpp

```diff
@@ -1,145 +1,196 @@
+    });
+  }
+
+  // See `void TraceSegment::inject(DictWriter& writer, const SpanData& span)`
+  // in dd-trace-cpp.
+  for (const auto style : finalized_config->injection_styles) {
+    switch (style) {
+      case dd::PropagationStyle::DATADOG:
+        result.push_back("x-datadog-trace-id");
+        result.push_back("x-datadog-parent-id");
+        result.push_back("x-datadog-sampling-priority");
+        result.push_back("x-datadog-origin");
+        result.push_back("x-datadog-tags");
+        break;
+      case dd::PropagationStyle::B3:
+        result.push_back("x-b3-traceid");
+        result.push_back("x-b3-spanid");
+        result.push_back("x-b3-sampled");
+        break;
+      case dd::PropagationStyle::W3C:
+        result.push_back("traceparent");
+        result.push_back("tracestate");
+        break;
+      case dd::PropagationStyle::NONE:
+        break;
     }
-    return {};
   }
 
-  const bool priority_sampling_enabled = true;
-  return ::datadog::opentracing::getPropagationHeaderNames(maybe_options->inject,
-                                                           priority_sampling_enabled);
+  return result;
 }
 
-string_view TracingLibrary::propagation_header_variable_name_prefix() {
+std::string_view TracingLibrary::propagation_header_variable_name_prefix() {
   return "datadog_propagation_header_";
 }
 
-string_view TracingLibrary::environment_variable_name_prefix() { return "datadog_env_"; }
+std::string_view TracingLibrary::environment_variable_name_prefix() { return "datadog_env_"; }
 
-string_view TracingLibrary::configuration_json_variable_name() { return "datadog_config_json"; }
+std::string_view TracingLibrary::configuration_json_variable_name() {
+  return "datadog_config_json";
+}
 
-string_view TracingLibrary::location_variable_name() { return "datadog_location"; }
+std::string_view TracingLibrary::location_variable_name() { return "datadog_location"; }
 
-string_view TracingLibrary::proxy_directive_variable_name() { return "datadog_proxy_directive"; }
+std::string_view TracingLibrary::proxy_directive_variable_name() {
+  return "datadog_proxy_directive";
+}
 
 namespace {
 
-std::string span_property(string_view key, const ot::Span &span) {
+class SpanContextJSONWriter : public dd::DictWriter {
+  nlohmann::json output_object_;
+
+ public:
+  SpanContextJSONWriter() : output_object_(nlohmann::json::object()) {}
+
+  void set(std::string_view key, std::string_view value) override {
+    std::string normalized_key;
+    std::transform(key.begin(), key.end(), std::back_inserter(normalized_key),
+                   header_transform_char);
+    output_object_[std::move(normalized_key)] = value;
+  }
+
+  nlohmann::json& json() { return output_object_; }
+};
+
+std::string span_property(std::string_view key, const dd::Span& span) {
   const auto not_found = "-";
 
   if (key == "trace_id") {
-    return span.context().ToTraceID();
+    return std::to_string(span.trace_id().low);
```

Comment:
Why only low?

Reply:
The question of how to render trace IDs in logs has caused problems in other tracers since the "switch" to 128-bit IDs.

As far as I know, trace IDs are reduced to 64-bit on the backend, and the log parsing does not understand 128-bit style IDs. I thought it best to print only that which will appear in the UI, while avoiding possible log correlation problems.